### DISCUSSION TITLE
No example of i2s input to Analog output (Build-in DAC pin 25/26 ESP32)?

### DISCUSSION THREAD

**Discussion by ShadGates** [pschatzmann/arduino-audio-tools#131] [Q&A] [answered]
Thanks for the potent library, been having a lot of fun playing with it.  

I'm wondering about the choice to not include an example of i2s input, output to built-in DAC pins?  It seems there's likely a technical reason for this, as you mention in another thread here that there are no examples of this.

I have some i2s amplifiers on order (MAX98357), but while waiting for them, setup up a TDA7052 amp circuit, which works for analog outputting examples.  I was trying to get a INMP441 i2s mic to output on the build-in DAC, but found no example and couldn't coerce the library into letting me do it.   

So, before I spent too much more time on it, I thought I'd ask if there's a technical reason you avoided this combination?

Thanks!

**Comment by pschatzmann** [comment]
Oh, there are plenty of examples have a look again. 
- Built in DAC examples ending with analog
- Built in ADC are using adc in their name. I might consider to rename them to analog as well to be consistent

**Reply by ShadGates** [reply]
Yes, I've played with most of them, thank you.   
My point was that none of them use an i2s source, like the INMP441, and output via analog.  

They're all programatic generated sources (MP3 or "generator").  

If there's no reason for it not to work, I'll continue in my efforts to convert the streams-generator-analog example to use an i2s source instead.

Thanks,
Shad

**Comment by pschatzmann** [comment, marked as answer]
Sure it will work. There is just no added value in providing examples for every possible combination of input and output....

**Reply by ShadGates** [reply]
Thanks
